roles/openshift_aws/tasks/wait_for_groups.yml

```diff
@@ -34,9 +34,17 @@
     msg: "{{ instancesout.results | sum(attribute='instances', start=[]) }}"
 
 - name: wait for ssh to become available
+  when: openshift_aws_wait_for_ssh | bool
   wait_for:
     port: 22
     host: "{{ item.public_ip_address }}"
     timeout: 300
     search_regex: OpenSSH
   with_items: "{{ instancesout.results | sum(attribute='instances', start=[]) }}"
+
+- name: wait for new nodes become ready
```

Comment:
Isn't node already in Ready state by that time? Previous task uses ssh to connect to it, so its definitely ready

Reply:
The previous task are check whether the instance is running well or not, it's not mean that when you run `oc get node <node>` is in "Ready" status, for some reason like the autoapprover pods is not running well, the csr not approved, or the sdn pods is crash status, the node is "NotReady", we need make sure all the node is in "Ready" status before we start drain the old nodes, or the pods will be in pending status.

Reply:
Oh, I see, its kubelet ready, not just "machine is ready". Makes sense, thanks